pkg/agent/baker.go

```diff
@@ -504,12 +504,18 @@ func getContainerServiceFuncMap(cs *api.ContainerService) template.FuncMap {
 			return cs.Properties.OrchestratorProfile.KubernetesConfig.NetworkPlugin == NetworkPluginKubenet
 		},
 		"NeedsContainerd": func() bool {
+			if profile != nil && profile.KubernetesConfig != nil {
+				return profile.KubernetesConfig.NeedsContainerd()
+			}
 			return cs.Properties.OrchestratorProfile.KubernetesConfig.NeedsContainerd()
 		},
 		"IsKataContainerRuntime": func() bool {
 			return cs.Properties.OrchestratorProfile.KubernetesConfig.ContainerRuntime == api.KataContainers
 		},
 		"IsDockerContainerRuntime": func() bool {
+			if profile != nil && profile.KubernetesConfig != nil {
+				return profile.KubernetesConfig.ContainerRuntime == api.Docker
```

Comment:
I don't think we use kubernetesConfig from agentpool in aks RP

Reply:
eventually we will have agentbaker caller pass in a simple struct with all the info we need, similar to  containerservice datamodel with one single pool so no need to care about multiple pools

Reply:
oh ok, that makes sense.

So a part of the initial containerd design is to tie the container runtime spec to the agent-pool level which will allow for currently existing clusters to be in a "hetereogenous" state i.e:
- customer adds a new agent pool to their existing MC after containerd has GA'ed or they're using the containerd preview flag
hence why I think templating here should

Though this really only applies to the `NeedsContainerd` flag. I added the additional agent pool profile checks in here as well mostly for symmetry.

Reply:
I can remove the change here

Reply:
hmm maybe a simpler way is to also just set the MC model in aks-rp to `containerd` in the above scenario so we can keep agentbaker simple.

Reply:
ok. i'm fine as long as we are aligned on the RP data model usage (i.e. we store agentpool container runtime config in agentpool table and pass it along to agentbaker)